analyzers/src/SonarAnalyzer.CSharp/Rules/ParametersCorrectOrder.cs

```diff
@@ -35,6 +35,16 @@ public sealed class ParametersCorrectOrder : ParametersCorrectOrderBase<Argument
     {
         private static readonly DiagnosticDescriptor rule =
             DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+
+        private static SyntaxToken? GetExpressionSyntaxIdentifier(ExpressionSyntax expressionSyntax) =>
+            expressionSyntax switch
+            {
+                IdentifierNameSyntax identifierNameSyntax => identifierNameSyntax.Identifier,
+                MemberAccessExpressionSyntax memberAccessExpressionSyntax => GetExpressionSyntaxIdentifier(memberAccessExpressionSyntax.Expression),
```

Comment:
We should not use recursion to all member access chain here, because doing something like `b.a` returns value of `a` for argument `a`.

Please check explicitly only for `.Name == "Value"` here. To be sure that we extract only `something.Value` => `"something"`.

Please add more test cases with some custom type having `Value` property.
Please add more test cases with `a.Something.b`

Reply:
Sure, it was a mistake, have added extended check for "Value"